test/src/main/java/org/apache/accumulo/test/functional/ThriftMaxFrameSizeIT.java

```diff
@@ -32,19 +37,20 @@
 import org.apache.thrift.TConfiguration;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
+import org.opentest4j.AssertionFailedError;
 
 public class ThriftMaxFrameSizeIT extends AccumuloClusterHarness {
 
   private ThriftServerType serverType;
 
-  @Override
-  protected Duration defaultTimeout() {
-    return Duration.ofMinutes(1);
-  }
-
   @Override
   public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    cfg.setNumTservers(1);
     cfg.setProperty(Property.GENERAL_RPC_SERVER_TYPE, serverType.name());
+    cfg.setProperty(Property.GENERAL_MAX_MESSAGE_SIZE,
+        Integer.toString(TConfiguration.DEFAULT_MAX_FRAME_SIZE));
+    cfg.setProperty(Property.TSERV_MAX_MESSAGE_SIZE,
+        Integer.toString(TConfiguration.DEFAULT_MAX_FRAME_SIZE));
```

Comment:
So, you're fundamentally altering the purpose of this test.

The purpose of this test was to make sure that we could pass messages larger than the 16MB max frame size that was the default. We are inherently relying on the fact that our default values for these on the server side are greater than this max, and we are passing messages that exceed the 16MB, to make sure that the configuration on the server-side worked as expected.

Your changes, however, make it so the server side is configured back to the default, making it indistinguishable from our server-side configuration being ignored, and verifying that the client hangs (except for SSL, which is weird), because it can *not* pass messages larger than the 16MB.

I don't think the changes to this IT are good here.

The situation I ran into when I was looking at this IT, was that I didn't just want to try to pass messages larger than 16MB... I wanted to try to pass messages larger than 100MB, to make sure that the changes in this PR correctly set the max message size when it set the max frame size. I was unable to modify the test to do that, because 100MB is huge, and I kept getting out of memory errors.

Rather than test that we can set it larger than 100MB, we can just test that we're overriding the default of 100MB to something like 20MB. We can then test that we can pass messages over 16MB, but less than 20MB, and we can verify that messages larger than 20MB won't work. But that doesn't tell us anything about whether they failed because they hit the max frame size or because they hit the max message size.

We also still might not be able to easily distinguish in the IT the reason why messages larger than 20MB won't work... it could be because of an OutOfMemoryError on the server, rather than exceeding the max message size. What I found interesting in these changes you made, was that you were basically testing that the client hung, and not that the client failed. Based on the EOFException change in TabletServerBatchReaderIterator, I had expected to see a hard failure, rather than a hang due to retries, but I guess we're hitting this max message issue in more places than that iterator and that change was a very specific change to address a related bug, rather than a general fix for retrying Thrift messages that are too big at the RPC layer.

Reply:
My test refactoring addresses pretty much all this. The cause of the larger messages not working is still not easily distinguishable, though. But, at least there's coverage for both cases now. I did spend quite a bit of time trying to find a good configuration size and sizes of messages to send, that would make sure the test passed okay. I'm still not sure why some smaller values don't work.